src/schema/rules/sidecars/asl_validation.yaml

```diff
@@ -0,0 +1,259 @@
+# Rules for ASL data that are not defined in tables.
+---
+
+# 157
+ASLLabelingDurationNiftiLength:
+    code: LABELING_DURATION_LENGTH_NOT_MATCHING_NIFTI
+    description: |
+        The number of values for 'LabelingDuration' for this file does not match the 4th dimension of the NIfTI header.
+        'LabelingDuration' is the total duration of the labeling pulse train, in seconds,
+        corresponding to the temporal width of the labeling bolus for `(P)CASL`.
+        In case all control-label volumes (or deltam or CBF) have the same `LabelingDuration`, a scalar must be
+        specified.
+        In case the control-label volumes (or deltam or cbf) have a different `LabelingDuration`,
+        an array of numbers must be specified, for which any `m0scan` in the timeseries has a `LabelingDuration` of
+        zero.
+        In case an array of numbers is provided, its length should be equal to the number of volumes specified in
+        `*_aslcontext.tsv`. Corresponds to DICOM Tag 0018,9258 `ASL Pulse Train Duration`.
+    level: error
+    selectors:
+    - suffix == "asl"
+    - sidecar contains "LabelingDuration"
+    - type(sidecar.LabelingDuration) == "array"
```

Comment:
Sorry for my sparse appearance at the meetings. So do we have a grammar and interpreter already? (Since in general looks like python but this absent `contains` operand throws me off, not sure why not to use available `in` with swap of the arguments)

Reply:
In YAML:

````YAML
- "LabelingDuration" in sidecar
```

raises an error because of YAML parsing rules. But we also don't insist that people always use `<` instead of `>`.